opencl/CMakeLists.txt

```diff
@@ -78,4 +78,24 @@ endif()
 FetchContent_MakeAvailable(ocl-icd)
 add_library(OpenCL-ICD ALIAS OpenCL)
 
+# OpenCL RT build requires headers from ICD loader, propagating location
+set(OpenCL_LOADER_DIR
+  ${ocl-icd_SOURCE_DIR} CACHE PATH "Path to OpenCL Loader")
+
+# TODO: Remove suppressing Werror.
+if(MSVC)
+  target_compile_options(OpenCL PRIVATE "-WX-")
+else()
+  target_compile_options(OpenCL PRIVATE "-Wno-error")
+endif()
+# Deploy OpenCL loader into package, this is required for xmain tc tests.
+# The deploy destination is different for OpenCL stand-alone product. See
+# opencl-intel/framework/CMakeLists.txt.
+if (NOT INTEL_OPENCL_STANDALONE_PRODUCT)
+  install(TARGETS OpenCL
+          ARCHIVE DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT OpenCL-ICD
+          LIBRARY DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT OpenCL-ICD
+          RUNTIME DESTINATION "bin" COMPONENT OpenCL-ICD)
+endif()
+
```

Comment:
I suggest we discuss if we want to deploy OpenCL CPU with DPC++ first. If so, we probably should update documentation, which currently says to install it separately. How do you envision DPC++ deployment process?

Reply:
First, I need to update the comment from line 91 to 93.  
I think we can update the document that we deploying OpenCL CPU (the open source version) with DPCPP++ by default.  If users want to use high performance OpenCL CPU RT, they can install it separately.

Reply:
Not just that, they need to remove the version deployed with DPC++, so I think it should go to a separate directory, which can be customized. If you put it to the same directory as DPC++, where only option would be physically to remove OpenCL CPU implementation.

Reply:
ok.  I will move the OCL ICD installation to OpenCL CPU RT CMake file.
The OpenCL CPU RT compilation in SYCLOS is optional.  By default it will only build the SYCL tool chain.  If users add -bldocl  option,  OpenCL CPU RT will be compiled and installed.

Reply:
> ok. I will move the OCL ICD installation to OpenCL CPU RT CMake file.

Sorry, I mean different thing. I see that OpenCL artifacts are installed to the same location and DPC++. This means that if we build both OpenCL CPU and DPC++, we can't configure DPC++ with pre-installed OpenCL CPU. Environment variables like `PATH` and `LD_LIBRARY_PATH` will point the directories with OpenCL CPU built from sources rather than installed in the system.

> The OpenCL CPU RT compilation in SYCLOS is optional. By default it will only build the SYCL tool chain. If users add -bldocl option, OpenCL CPU RT will be compiled and installed.

Okay. We need to add instructions how to build OpenCL CPU to Get Started Guide guide.

What about CI? I expect you want to have CI validation for OpenCL. Right?

Reply:
Generally OpenCL Runtime is detected and loaded through full path in registry key (windows) or definition in ICD driver file or OCL_ICD_FILENAMES environment. It does not depends on PATH or LD_LIBRARY_PATH env.  So user can configure DPC++ with pre-installed OpenCL CPU runtime following the same configuration as current document in SYCLOS. 

We will create YML file for OpenCL CPU RT to enable CI test.  I will ask you to review when we finish it.

Reply:
> Generally OpenCL Runtime is detected and loaded through full path in registry key (windows) or definition in ICD driver file or OCL_ICD_FILENAMES environment. It does not depends on PATH or LD_LIBRARY_PATH env. So user can configure DPC++ with pre-installed OpenCL CPU runtime following the same configuration as current document in SYCLOS.

The "entry point" library loaded by ICD loads the rest of shared libraries using pre-defined relative paths instead of relying on LD_LIBRARY_PATH? If so, that resolves my concern.

Reply:
For OpenCL CPU RT,  the rest of shared libraries (TBB libraries are an exception) are loaded from the same folder of entry point library.  They are not loaded from LD_LIBRARY_PATH.  TBB libraries are loaded for LD_LIBRARY_PATH.